activerecord/test/cases/relation/where_test.rb

```diff
@@ -405,6 +405,12 @@ def test_where_with_relation_on_has_one_association
       assert_equal author_addresses(:david_address), author_address
     end
 
+    def test_has_many_with_plural_association_name
+      author = authors(:david)
+      author_address = AuthorAddress.where(authors: Author.where(id: author.id)).first
```

Comment:
This is testing the functionality we want, but for the sake of being a bit more illustrative of a plausible use case, perhaps instead of the id reference here, we'd be better off with name?

i.e., the test would be producing a list of addresses of authors named David (of which our test data happens to have one)

(Just aiming that, with the same data set, we have a query that's a little more believable that a developer might intuitively reach for the plural)